### PR TITLE
Update to new CBA event system

### PR DESCRIPTION
**Pull request by PabstMirror**
Ref https://github.com/acemod/ACE3/pull/3815

### CODE REVIEW

File: addons/headless/XEH_postInit.sqf [hunk 1]
```diff
@@ -3,14 +3,14 @@
...
+["ace_settingsInitialized", {
     if (isServer) then {
         // Add disconnect EH if HC transferring enabled
         if (GVAR(Enabled)) then {
             addMissionEventHandler ["HandleDisconnect", {_this call FUNC(handleDisconnect)}];
         };
     } else {
         // Register HC (this part happens on HC only)
-        ["ACE_HeadlessClientJoined", [player]] call ACEFUNC(common,globalEvent);
+        [QGVAR(HeadlessClientJoined), [player]] call CBA_fnc_globalEvent;
```
Comment (jonpas) [inline addons/headless/XEH_postInit.sqf]: Lowercase `h`.

File: addons/headless/XEH_preInit.sqf [hunk 2]
```diff
@@ -7,7 +7,7 @@ ADDON = false;
 if (isServer) then {
     GVAR(headlessClients) = [];
     GVAR(inRebalance) = false;
-    ["ACE_HeadlessClientJoined", FUNC(handleConnectHC)] call ACEFUNC(common,addEventHandler);
+    [QGVAR(HeadlessClientJoined), FUNC(handleConnectHC)] call CBA_fnc_addEventHandler);
```
Comment (jonpas) [inline addons/headless/XEH_preInit.sqf]: Lowercase `h`.

File: addons/headless/config.cpp [hunk 3]
```diff
@@ -16,3 +16,7 @@ class CfgPatches {
 #include "ACE_Settings.hpp"
 #include "CfgEventHandlers.hpp"
 #include "CfgVehicles.hpp"
+
+class ACE_newEvents {
+    ACE_HeadlessClientJoined = QGVAR(HeadlessClientJoined);
```
Comment (jonpas) [inline addons/headless/config.cpp]: Lowercase `h`.

File: addons/headless/XEH_preInit.sqf (hunk 2 above)
Comment (Glowbal) [inline addons/headless/XEH_preInit.sqf]: Excessive `)`

**Comment by Glowbal** [comment]
Perfect, Thanks for making this @PabstMirror :+1:

**Comment by jonpas** [comment]
:+1:

**Pull request merged**